controller/tests/late_lock.rs

```diff
@@ -0,0 +1,163 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Tests and experimentations with late locking
+#[macro_use]
+extern crate log;
+extern crate grin_wallet_controller as wallet;
+extern crate grin_wallet_impls as impls;
+extern crate grin_wallet_libwallet as libwallet;
+
+use self::libwallet::{InitTxArgs, Slate};
+use impls::test_framework::{self, LocalWalletClient};
+use std::sync::atomic::Ordering;
+use std::thread;
+use std::time::Duration;
+
+#[macro_use]
+mod common;
+use common::{clean_output_dir, create_wallet_proxy, setup};
+
+/// self send impl
+fn late_lock_test_impl(test_dir: &'static str) -> Result<(), libwallet::Error> {
+	// Create a new proxy to simulate server and wallet responses
+	let mut wallet_proxy = create_wallet_proxy(test_dir);
+	let chain = wallet_proxy.chain.clone();
+	let stopper = wallet_proxy.running.clone();
+
+	// Create a new wallet test client, and set its queues to communicate with the
+	// proxy
+	create_wallet_and_add!(
+		client1,
+		wallet1,
+		mask1_i,
+		test_dir,
+		"wallet1",
+		None,
+		&mut wallet_proxy,
+		false
+	);
+	let mask1 = (&mask1_i).as_ref();
+	create_wallet_and_add!(
+		client2,
+		wallet2,
+		mask2_i,
+		test_dir,
+		"wallet2",
+		None,
+		&mut wallet_proxy,
+		false
+	);
+	let mask2 = (&mask2_i).as_ref();
+
+	// Set the wallet proxy listener running
+	thread::spawn(move || {
+		if let Err(e) = wallet_proxy.run() {
+			error!("Wallet Proxy error: {}", e);
+		}
+	});
+
+	// add some accounts
+	wallet::controller::owner_single_use(Some(wallet1.clone()), mask1, None, |api, m| {
+		api.create_account_path(m, "mining")?;
+		api.create_account_path(m, "listener")?;
+		Ok(())
+	})?;
+
+	// add some accounts
+	wallet::controller::owner_single_use(Some(wallet2.clone()), mask2, None, |api, m| {
+		api.create_account_path(m, "account1")?;
+		api.create_account_path(m, "account2")?;
+		Ok(())
+	})?;
+
+	// Get some mining done
+	{
+		wallet_inst!(wallet1, w);
+		w.set_parent_key_id_by_name("mining")?;
+	}
+	{
+		wallet_inst!(wallet2, w);
+		w.set_parent_key_id_by_name("account1")?;
+	}
+
+	let bh = 10u64;
+	let _ =
+		test_framework::award_blocks_to_wallet(&chain, wallet1.clone(), mask1, bh as usize, false);
+
+	let mut slate = Slate::blank(2, false);
+	let amount = 60_000_000_000;
+
+	wallet::controller::owner_single_use(Some(wallet1.clone()), mask1, None, |sender_api, m| {
+		let args = InitTxArgs {
+			src_acct_name: Some("mining".to_owned()),
+			amount,
+			minimum_confirmations: 2,
+			max_outputs: 500,
+			num_change_outputs: 1,
+			selection_strategy_is_use_all: false,
+			late_lock: Some(true),
+			..Default::default()
+		};
+		let slate_i = sender_api.init_send_tx(m, args)?;
+		println!("S1 SLATE: {}", slate_i);
+		slate = client1.send_tx_slate_direct("wallet2", &slate_i)?;
+		println!("S2 SLATE: {}", slate);
+
+		// Note we're not doing this here
+		// sender_api.tx_lock_outputs(m, &slate)?;
```

Comment:
Maybe remove the commented code and replace with a comment to explain what is happening?
Or maybe this should just be removed entirely?